pkg/schema/tracker.go

```diff
@@ -329,3 +340,182 @@ func (tr *Tracker) CreateTableIfNotExists(table *filter.Table, ti *model.TableIn
 func (tr *Tracker) GetSystemVar(name string) (string, bool) {
 	return tr.se.GetSessionVars().GetSystemVar(name)
 }
+
+// GetToIndexInfo gets downstream PK Index.
+// note. this function will init toIndexes.
+func (tr *Tracker) GetToIndexInfo(db, table string, originTi *model.TableInfo, tctx *tcontext.Context, task string, tidbConn *conn.BaseConn) (*model.IndexInfo, error) {
+	if tr.toIndexes == nil {
+		tr.toIndexes = make(map[string]map[string]*ToIndexes)
+	}
+	if dbindexes := tr.toIndexes[db]; dbindexes == nil {
+		dbindexes = make(map[string]*ToIndexes)
+		tr.toIndexes[db] = dbindexes
+	}
+	index := tr.toIndexes[db][table]
+	if index == nil {
+		log.L().Info(fmt.Sprintf("DownStream schema tracker init: %s.%s", db, table))
+		index = &ToIndexes{
+			schemaName: db,
+			tableName:  table,
+			pks:        make([]string, 0),
+			uks:        make([]string, 0),
+			// uksIsNull:  make([]bool, 0),
+		}
+		// tctx := tcontext.NewContext(ctx, log.With(zap.String("component", "schema-tracker"), zap.String("task", task)))
+		rows, err := tidbConn.QuerySQL(tctx, fmt.Sprintf("SHOW INDEX FROM %s FROM %s", table, db))
+		if err != nil {
+			return nil, err
+		}
+
+		cols, err := rows.Columns()
+		if err != nil {
+			return nil, err
+		}
+		// the column of show statement is too many, so make dynamic values for scan
+		values := make([][]byte, len(cols))
+		scans := make([]interface{}, len(cols))
+		for i := range values {
+			scans[i] = &values[i]
+		}
+
+		for rows.Next() {
+			if err3 := rows.Scan(scans...); err3 != nil {
+				return nil, err3
+			}
+
+			// Key_name -- 2, Column_name -- 4, Null -- 9
+			nonUnique := string(values[1]) // 0 is UK
+			keyName := string(values[2])   // pk is PRIMARY
+			columName := string(values[4])
+			// isNull := string(values[9]) // Null is YES
+
+			if strings.EqualFold(keyName, "PRIMARY") {
```

Comment:
is EqualFold needed? or just use `==` op?

Reply:
yes，i think it can use '=='